### PR TITLE
[confighttp] Allow compression list for a server to be overridden

### PR DESCRIPTION
**Pull request by jpkrohling**
- **[confighttp] Allow compression list for a server to be overridden**
- **changelog**

### CODE REVIEW

File: .chloggen/jpkroehling_allow-compression-list-to-be-overridden.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Allow the compression list to be overridden
+
+# One or more tracking issues or pull requests related to the change
+issues: []
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: Allows Collector adminitrators to control which compression algorithms to enable for HTTP-based receivers.
```
Comment (codeboten) [inline .chloggen/jpkroehling_allow-compression-list-to-be-overridden.yaml:18]:
```suggestion
subtext: Allows Collector administrators to control which compression algorithms to enable for HTTP-based receivers.
```

File: config/confighttp/confighttp.go [hunk 2]
```diff
@@ -280,6 +281,9 @@ type ServerConfig struct {
 	// Additional headers attached to each HTTP response sent to the client.
 	// Header values are opaque since they may be sensitive.
 	ResponseHeaders map[string]configopaque.String `mapstructure:"response_headers"`
+
+	// Compression configures the list of compression algorithms the server can accept. Default: ["", "gzip", "zstd", "zlib", "snappy", "deflate"]
+	Compression []string `mapstructure:"compression"`
```
Comment (codeboten) [inline config/confighttp/confighttp.go:286]: Please add an entry in the README for the new option

File: config/confighttp/confighttp.go [hunk 3]
```diff
@@ -280,6 +281,9 @@ type ServerConfig struct {
 	// Additional headers attached to each HTTP response sent to the client.
 	// Header values are opaque since they may be sensitive.
 	ResponseHeaders map[string]configopaque.String `mapstructure:"response_headers"`
+
+	// Compression configures the list of compression algorithms the server can accept. Default: ["", "gzip", "zstd", "zlib", "snappy", "deflate"]
```
Comment (codeboten) [inline config/confighttp/confighttp.go:285]: I'm not sure if the name is clear enough for what this option is, what do you think of something like `CompressionAlgorithms` or `SupportedCompression`?

File: config/confighttp/confighttp.go [hunk 4]
```diff
@@ -31,6 +31,7 @@ import (
 
 const headerContentEncoding = "Content-Encoding"
 const defaultMaxRequestBodySize = 20 * 1024 * 1024 // 20MiB
+var defaultCompressionList = []string{"", "gzip", "zstd", "zlib", "snappy", "deflate"}
```
Comment (codeboten) [inline config/confighttp/confighttp.go:34]:
```suggestion
var defaultCompressionAlgorithms = []string{"", "gzip", "zstd", "zlib", "snappy", "deflate"}
```

File: config/confighttp/confighttp.go [hunk 5]
```diff
@@ -345,7 +349,11 @@ func (hss *ServerConfig) ToServer(_ context.Context, host component.Host, settin
 		hss.MaxRequestBodySize = defaultMaxRequestBodySize
 	}
 
-	handler = httpContentDecompressor(handler, hss.MaxRequestBodySize, serverOpts.errHandler, serverOpts.decoders)
+	if hss.CompressionAlgorithms == nil {
+		hss.CompressionAlgorithms = defaultCompressionList
```
Comment (codeboten) [inline config/confighttp/confighttp.go:353]:
```suggestion
		hss.CompressionAlgorithms = defaultCompressionAlgorithms
```

**Review by codeboten — APPROVED** [review]

**Comment by jpkrohling** [comment]
@codeboten , is this OK to be merged?

**Review by codeboten — APPROVED** [review]

**Pull request merged**